terragrunt/aws/alert_compromise/lambda.tf

```
@@ -0,0 +1,58 @@
+# Local variables to get the api lambda function group name and arn
+locals {
+  target_log_group_name = "/aws/lambda/${var.api_function_name}"
+  target_log_group_arn  = "arn:aws:logs:${var.region}:${var.account_id}:log-group:${local.target_log_group_name}"
+}
+
+# This data block creates an archive file of the broadcast_alert.py script
+# to be used in the AWS Lambda function.
+data "archive_file" "broadcast_alert" {
+  type        = "zip"
+  source_file = "${path.module}/functions/broadcast_alert.py"
+  output_path = "/tmp/broadcast_alert.py.zip"
+}
+
+# Lambda function that will broadcast the alert to the SNS topic
+resource "aws_lambda_function" "broadcast_alert" {
+  function_name                  = var.function_name
+  description                    = "Notify other teams that their API keys have been compromised."
+  role                           = aws_iam_role.group_broadcast_alert_role.arn
+  handler                        = "broadcast_alert.lambda_handler"
+  runtime                        = "python3.9"
```

Could you bump this if your code supports it:
```suggestion
  runtime                        = "python3.11"
```